src/main/java/bio/terra/common/LockOperation.java

```diff
@@ -27,20 +28,55 @@ public enum LockOperation {
    * @return additional guidance for users investigating a failed lock operation, usually as part of
    *     a failed job / flight.
    */
-  public List<String> getErrorDetails() {
+  public List<String> getErrorDetails(ResourceLocks conflictingLocks) {
     var template =
         "A failure to obtain %s on a resource likely means that it's already %s by another process.";
     if (description != null && conflict != null) {
+      if (conflictingLocks != null && !isEmpty(conflictingLocks)) {
+        template =
+            template
+                + " Conflicting lock(s): %s. You can remove a lock by using the unlock API endpoint with the ID of the resource (Snapshot or Dataset), the name of the lock, and forceUnlock set to false.";
```

Comment:
It looks like there might be cases where the `%s` would not be formatted, for example if `this != LOCK_EXCLUSIVE`

Reply:
I don't think I see what you're saying. This is my understanding. if this is not exclusive, it must be a shared lock(since this exists within a check that description and conflict are not null meaning it isn't an unlock). And since it also exists within a check that there is a conflicting lock, it must be an exclusive lock. Is that right?